sklearn/linear_model/coordinate_descent.py

```diff
@@ -692,7 +690,7 @@ def fit(self, X, y, check_input=True):
         if self.selection not in ['cyclic', 'random']:
             raise ValueError("selection should be either random or cyclic.")
 
-        if not self.warm_start or self.coef_ is None:
+        if not self.warm_start or getattr(self, "coef_", None) is None:
```

Comment:
There is some failure if I do that in sklearn/decomposition/tests/test_dict_learning.py that I haven't fully understood.